aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -966,10 +978,53 @@ void scaled_gemm(
       &heuristicResult,
       &returnedResult));
   if (returnedResult == 0) {
+#ifndef USE_ROCM
     TORCH_CUDABLAS_CHECK(CUBLAS_STATUS_NOT_SUPPORTED);
+#else
+    // hipblaslt might be able to recover by returning all algos
+    std::vector<hipblasLtMatmulHeuristicResult_t> all_algos;
+    TORCH_CUDABLAS_CHECK(hipblaslt_ext::getAllAlgos(
+        ltHandle,
+        hipblaslt_ext::GemmType::HIPBLASLT_GEMM,
+        _cublasOpFromChar(transa),
+        _cublasOpFromChar(transb),
+        HIPTOLT(ScalarTypeToCudaDataType(mat1_dtype)),
+        HIPTOLT(ScalarTypeToCudaDataType(mat2_dtype)),
+        // C is nullptr and beta=0, so set to something reasonable. See above.
+        //HIPTOLT(ScalarTypeToCudaDataType(bias_dtype)),
+        HIPTOLT(ScalarTypeToCudaDataType(result_dtype)),
+        HIPTOLT(ScalarTypeToCudaDataType(result_dtype)),
+        HIPCOMPTOLT(CUBLAS_COMPUTE_32F),
+        all_algos));
+    if (all_algos.size() == 0) {
+      TORCH_CUDABLAS_CHECK(CUBLAS_STATUS_NOT_SUPPORTED);
+    }
+    // pick first valid solution
+    bool found = false;
+    for (size_t i = 0; i < all_algos.size(); i++) {
+        size_t ret_workspace_size = 0;
+        auto is_valid_status = hipblaslt_ext::matmulIsAlgoSupported(
+                ltHandle,
+                computeDesc.descriptor(),
+                &alpha_val,
+                Adesc.descriptor(),
+                Bdesc.descriptor(),
+                &beta_val,
+                Cdesc.descriptor(),
+                Ddesc.descriptor(),
+                all_algos[i].algo,
+                ret_workspace_size);
+        if (is_valid_status == HIPBLAS_STATUS_SUCCESS) {
+            if (ret_workspace_size <= workspaceSize) {
+                heuristicResult = all_algos[i];
+                found = true;
+                break;
+            }
+        }
+    }
```

Comment:
Ideally the above code can be encapsulated into the same `cublasLtMatmulAlgoGetHeuristic` API, so we will not diverge here.

Reply:
Currently hipblasLtMatmulAlgoGetHeuristic returns 0 solutions.  That's why I added this fallback.